test/service/test_manifest.py

```diff
@@ -184,6 +197,11 @@ def test_manifest(self):
                         self.assertEqual(200, response.status_code, 'Unable to download manifest')
                         self._assert_tsv(expected, response)
 
+            # The following assertion fails due to ManifestService attempt to use a cached manifest rather a
+            # freshly generated one.
+            response = self._get_manifest('compact', {})
+            self.assertEqual(200, response.status_code, 'Unable to download manifest')
```

Comment:
This does not assert that the response was freshly generated.

 I think your proof works but only because I understand the details here and know how to interpret the KeyError. No one else is going to be able to follow your thought process in the future. That's why this change needs to be dropped which you anticipated by labelling the commit as such.

But can you assert the freshness of the manifest in a way that's accurately documented and that can remain in the codebase? I would compare the two signed URLs that come back from the service. Maybe there is a difference between them that indicates freshness?

I have seen evidence for the effectiveness of your patch in the `run()` override. If the above can't be achieved, I'd be happy to move on.

Reply:
@hannes-ucsc  
No information regarding cache utilization can be extracted from manifest generation response. The generator will use different pre-signed urls regardless of whether a cached manifest is used or not. Also, manifest object_key/file_name will always be the same provided that the `filters`, `format` and `enable_multipart_upload` boolean are consistent. 
However, this could change if you decide it is better to separate the object_key generation from the pre-signed url generation, currently in the same method. With a separation in place, we could distinguish methods from taking a 'utilize cached manifest' path to a 'generate manifest' one, removing the need for the patch.

Reply:
I think 639c075 addresses this nicely.